File: spring/spring-webflux/webflux-study/ch09/hacking-spring-boot-ch9-reactive-quick-demo/src/main/java/com/greglturnquist/hackingspringboot/reactive/UserController.java
package com.greglturnquist.hackingspringboot.reactive;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Mono;

@RestController
@RequiredArgsConstructor
public class UserController {

    private final UserRepository repository;

    @GetMapping("/users/{name}")
    Mono<User> findUser(@PathVariable String name) {
        return repository.findByName(name);
    }
}

File: kotlin/kotlin-in-depth-book/src/main/kotlin/com/example/kotlinindepthbook/controller/UserController.kt
package com.example.kotlinindepthbook.controller

import com.example.kotlinindepthbook.domain.User
import com.example.kotlinindepthbook.service.UserService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/users")
class UserController(private val userService: UserService) {

    @GetMapping
    fun searchUser(username: String): User? {
        return userService.searchUserByUsername(username)
    }
}
